repository/src/main/java/org/alfresco/repo/content/AbstractRoutingContentStore.java

```diff
@@ -449,4 +451,51 @@ public boolean isStorageClassesSupported(Set<String> storageClasses)
         }
         return supportedStorageClasses;
     }
+
+    @Override
+    public void updateStorageClasses(String contentUrl, Set<String> storageClasses,
+        Map<String, Object> parameters)
+    {
+        List<ContentStore> stores = getAllStores();
+        for (ContentStore store : stores)
+        {
+            if (store.isStorageClassesSupported(storageClasses))
+            {
+                store.updateStorageClasses(contentUrl, storageClasses, parameters);

```

Comment:
There might be cases when the `storageClasses` are supported but the content is not in this store and the update will fail.

As it is the code will rethrow the exception further on.
I'm just think out loud whether or not we should address those exceptions in a way or another.